### ISSUE TITLE
Reindexing performance degrades non-linearly

### ISSUE THREAD

**Issue by bhalsey** [FusionAuth/fusionauth-issues#2896]
### Description
A test instance with 460M entities took over 4 days to complete on a 3 node cluster running FusionAuth 1.52.1.  The performance slowed significantly after 300M entities were indexed.  More work is required to clearly identify the bottleneck.  And more work is required to find mitigations, such as doubling the cluster size.

<img width="795" alt="Screenshot 2024-10-09 at 10 37 05 AM" src="https://github.com/user-attachments/assets/6b9ac3ff-9973-40b2-8ab7-f7b90b3118fc">

**Comment by robotdan** [comment]
There is not a lot of information here. Is the thesis that this is an problem with FusionAuth, or that we just need to scale Elasticsearch and the relational database adequately? 

Ideally we would only open public GH issues for things that need work in FusionAuth.

**Comment by bhalsey** [comment]
Agreed that more work is needed.  The outcome could simply be guidance on sizing a cluster.  Or it could entail changes to how FusionAuth manages reindexing, such as modifying the index refresh interval.
